src/components/Metadata/Metadata.tsx

```diff
@@ -39,69 +39,74 @@ const propsCompare = (_, newProps: MetadataProps) => {
 
 // TODO: Refactor loading state logic
 // TODO: Split "Metadata" component into "Metadata" and "PrivateMetadata" components
-export const Metadata: React.FC<MetadataProps> = memo(
-  ({ data, onChange, isLoading, readonly = false, hidePrivateMetadata = false, ...props }) => {
-    const change = (event: ChangeEvent, isPrivate: boolean) => {
-      const { action, field, fieldIndex, value } = parseEventData(event);
-      const key = getDataKey(isPrivate);
-      const dataToUpdate = data[key];
+export const MetadataNoMemo: React.FC<MetadataProps> = ({
+  data,
+  onChange,
+  isLoading,
+  readonly = false,
+  hidePrivateMetadata = false,
+  ...props
+}) => {
+  const change = (event: ChangeEvent, isPrivate: boolean) => {
+    const { action, field, fieldIndex, value } = parseEventData(event);
+    const key = getDataKey(isPrivate);
+    const dataToUpdate = data[key];
 
-      onChange({
-        target: {
-          name: key,
-          value:
-            action === EventDataAction.update
-              ? updateAtIndex(
+    onChange({
+      target: {
+        name: key,
+        value:
+          action === EventDataAction.update
+            ? updateAtIndex(
+                {
+                  ...dataToUpdate[fieldIndex],
+                  key: field === EventDataField.name ? value : dataToUpdate[fieldIndex].key,
+                  value: field === EventDataField.value ? value : dataToUpdate[fieldIndex].value,
+                },
+                dataToUpdate,
+                fieldIndex,
+              )
+            : action === EventDataAction.add
+              ? [
+                  ...dataToUpdate,
                   {
-                    ...dataToUpdate[fieldIndex],
-                    key: field === EventDataField.name ? value : dataToUpdate[fieldIndex].key,
-                    value: field === EventDataField.value ? value : dataToUpdate[fieldIndex].value,
+                    key: "",
+                    value: "",
                   },
-                  dataToUpdate,
-                  fieldIndex,
-                )
-              : action === EventDataAction.add
-                ? [
-                    ...dataToUpdate,
-                    {
-                      key: "",
-                      value: "",
-                    },
-                  ]
-                : removeAtIndex(dataToUpdate, fieldIndex),
-        },
-      });
-    };
+                ]
+              : removeAtIndex(dataToUpdate, fieldIndex),
```

Comment:
suggestion: I'd extract this to external function and create at least map action name to handler and then use it like that `fun getActionType(type) => map[actionType]`

Reply:
I'll just leave this component as is, since I didn't change anything here (other than export non-memoized component) and these were just formatting changes - I don't want to break anything here by accident 😄